File: TestSample/java/src/main/java/helloworld/speakers/KoSpeaker.java
package helloworld.speakers;

import helloworld.Person;

public final class KoSpeaker extends Person {
    public KoSpeaker(String name) {
        super(name, "ko");
    }

    @Override
    public String speak() {
        return "안녕, 세상!";
    }
}

File: TestSample/rust/src/speakers/ko_speaker.rs
use crate::person::{Person, PersonBase};

pub struct KoSpeaker {
    base: PersonBase,
}

impl KoSpeaker {
    pub fn new(name: &str) -> Self {
        Self { base: PersonBase::new(name, "ko") }
    }
}

impl Person for KoSpeaker {
    fn name(&self) -> &str { &self.base.name }
    fn language(&self) -> &str { &self.base.language }
    fn speak(&self) -> String { String::from("안녕, 세상!") }
}
